apps/extension/src/routes/page/restore-password.tsx

```diff
@@ -0,0 +1,40 @@
+import { BackIcon, Card, CardContent, CardDescription, CardHeader, CardTitle } from 'ui/components';
+import { cn } from 'ui/lib/utils';
+import { FadeTransition, ImportForm } from '../../components';
+import { useStore } from '../../state';
+import { importSelector } from '../../state/seed-phrase/import';
+import { passwordSelector } from '../../state/password';
+import { useEffect } from 'react';
+
+export const RestorePassword = () => {
+  const { phrase } = useStore(importSelector);
+  const { hashedPassword } = useStore(passwordSelector);
+
+  const replaceUrl = () => window.location.replace(chrome.runtime.getURL('page.html'));
+
+  useEffect(() => {
+    // redirect to index route after login from other tabs or popup, needed to clean up url
+    if (hashedPassword) replaceUrl();
+  }, [hashedPassword]);
```

Comment:
Why would someone end up on this page if they already had a password?

Reply:
User open tab and enter password to popup